pennylane_lightning/src/gates/cpu_kernels/avx_common/ApplyGeneratorIsingYY.hpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file
+ * Defines IsingYY generator
+ */
+#pragma once
+#include "AVXUtil.hpp"
+#include "BitUtil.hpp"
+#include "Permutation.hpp"
+#include "Util.hpp"
+
+#include <immintrin.h>
+
+#include <complex>
+
+namespace Pennylane::Gates::AVXCommon {
+
+template <typename PrecisionT, size_t packed_size>
+struct ApplyGeneratorIsingYY {
+    using Precision = PrecisionT;
+    using PrecisionAVXConcept =
+        typename AVXConcept<PrecisionT, packed_size>::Type;
+
+    constexpr static size_t packed_size_ = packed_size;
+    constexpr static bool symmetric = true;
+
+    template <size_t rev_wire0, size_t rev_wire1>
+    static auto applyInternalInternal(std::complex<PrecisionT> *arr,
+                                      size_t num_qubits,
+                                      [[maybe_unused]] bool adj) -> PrecisionT {
+        using namespace Permutation;
+        constexpr static auto perm = compilePermutation<Precision, packed_size>(
+            flip(flip(identity<packed_size>(), rev_wire0), rev_wire1));
```

Comment:
Functions like `flip` and others in the `Permutation` header do not seem to have any documentation explaining their intended use. Can these be added?

Reply:
Follow-up on the above comment:
```
Functions like flip and others in the Permutation header do not seem to have any
 documentation explaining their intended use. Can these be added?
```

Reply:
Yeap. I just added some explanation here.